agents/openstack/fence_openstack.py

```diff
@@ -11,12 +11,15 @@
 from fencing import fail_usage, is_executable, run_command, run_delay
 
 try:
-        from keystoneclient.v3 import client as ksclient
+        from novaclient import client as novaclient
+        from keystoneauth1 import session as ksc_session
+        from keystoneauth1 import loading
+        legacy_import = False
+except ImportError:
         from novaclient import client as novaclient
         from keystoneclient import session as ksc_session
         from keystoneclient.auth.identity import v3
```

Comment:
I think you meant to put `legacy_import=True` here.

Reply:
The `legacy_import=True` is in the `ImportError` section below

Reply:
It needs some extra nesting, so I'll take care of that after merging.